### PR TITLE
Define the differentiable variables for NN operator

### PR DESCRIPTION
**Pull request by rajeevnalawadi**
Pytorch has 3 specific modes for quantization
For training case the “Quantization aware training” QAT is applicable as described in the below linked page
https://pytorch.org/docs/stable/quantization.html#quantization-workflows

Snippet from the page under Quantization workflows & QAT (#3) “…Computations will take place in FP32 but with values clamped and rounded to simulate the effects of INT8 quantization…”

Code for checking QAT
https://pytorch.org/tutorials/advanced/static_quantization_tutorial.html

### CODE REVIEW

File: onnx/defs/nn/defs.cc [hunk 1]
```diff
@@ -1191,10 +1271,6 @@ void convTransposeShapeInference(InferenceContext& ctx) {
     if (pads.size() != n_input_dims * 2) {
       fail_shape_inference("Attribute pads has incorrect size");
     }
-    const auto* auto_pad_attr = ctx.getAttribute("auto_pad");
```
Comment (wschin) [inline onnx/defs/nn/defs.cc:1194]: Why is this removed?

Comment (rajeevnalawadi) [inline onnx/defs/nn/defs.cc:1194]: Manual add

File: docs/Operators.md [hunk 2]
```diff
@@ -3283,20 +3283,20 @@ This version of the operator has been available since version 10 of the default
 #### Inputs (2 - 4)
 
 <dl>
-<dt><tt>x</tt> : T1</dt>
+<dt><tt>x</tt> (non-differentiable) : T1</dt>
```
Comment (wschin) [inline docs/Operators.md:3286]: Why is it non-differentiable? To speed up the merge, we probably should split this PR into two PRs (one for normal ops and the other one for quantized ops)?

File: docs/Operators.md [hunk 3]
```diff
@@ -12744,30 +12744,30 @@ This version of the operator has been available since version 10 of the default
 #### Inputs (8 - 9)
 
 <dl>
-<dt><tt>x</tt> : T1</dt>
+<dt><tt>x</tt> (differentiable) : T1</dt>
```
Comment (wschin) [inline docs/Operators.md:12747]: This is differentiable but in `ConvInteger` the input `X` isn't.

File: docs/Operators.md (hunk 2 above)
Comment (rajeevnalawadi) [reply docs/Operators.md]: Removed the differentiable variables for QlinearConv and ConvInteger operators

File: docs/Operators.md (hunk 3 above)
Comment (rajeevnalawadi) [reply docs/Operators.md]: Removed the differentiable variables for QlinearConv and ConvInteger operators

**Review by wschin — APPROVED** [review]
Thanks for the hard work!
